File: 6_multiplevel_inheritance.py
class Vehicle:
    def __init__(self, brand):
        self.brand = brand

class Car(Vehicle):
    def __init__(self, brand, model):
        super().__init__(brand)
        self.model = model

class car_electric(Car):
    def __init__(self, brand, model, battery_capacity):
        super().__init__(brand, model)
        self.battery_capacity = battery_capacity

    def display(self):
        print(f"The car Brand: {self.brand}, Model: {self.model}, Battery Capacity: {self.battery_capacity} kWh")

def main():
    my_car = car_electric("Tesla", "Model S", 200)

    my_car.display()

if __name__ == "__main__":
    main()

File: 6_multiplevel_inheritance.cpp
#include <bits/stdc++.h>
using namespace std;

class Vehicle
{
protected:
    string brand;

public:
    Vehicle(string b)
    {
        brand = b;
    }
};

class Car : public Vehicle
{
protected:
    string model;

public:
    Car(string b, string m) : Vehicle(b), model(m) {}
};

class car_electric : public Car
{
protected:
    int batteryCapacity;

public:
    car_electric(string b, string m, int capacity) : Car(b, m), batteryCapacity(capacity) {}

    void display()
    {
        cout << "The car Brand: " << brand << ", Model: " << model << ", Battery Capacity: " << batteryCapacity << " kWh" << endl;
    }
};

int main()
{

    car_electric myCar("Tesla", "Model S", 200);

    myCar.display();

    return 0;
}
